### PR TITLE
add duplicates components for results list

### PR DESCRIPTION
**Pull request by gleibfried**
I have added the 3 duplicates components (`sq-result-duplicates`, `sq-result-duplicates-spacer`, `sq-result-duplicates-list`) for the results but I'm not familiar with how it works.

To make some tests I've updated the `testDuplicates` method parameter to always return true, which makes some behavior for the components but nothing crazy with something like that when having the 3 at the top of the result item:

image

(duplicates-list remain empty, the spacer is displayed for all records but the first, but I guess we don't expect it looks like that, and duplicates "seems" to work while displaying "19 near duplicates" only for the first one)

One thing I haven't been able to do from how sba-internal is doing is the `duplicate` class we apply to the record item so the CSS associated from search.scss is applied.
image

Here we don't handle the list ourselves but with `uib-zone` so I don't seem to have access and control to do this.

If you have ideas I'm open!

### CODE REVIEW

**Comment by ericleib** [comment]
ça me va d'intégrer `sq-result-duplicates` et `sq-result-duplicates-list`, mais ça me semble compliqué d'inclure aussi `sq-result-duplicates-spaces`, parce que le visuel dépend trop du CSS et de la façon dont les autres composants sont affichés. Pareil, les règles CSS sur `.duplicate` sont trop spécifiques à vanilla-search.

**Comment by gleibfried** [comment]
@ericleib Voilà j'ai retiré, ce qui reste semble bon et suffisant ?

File: projects/vanilla/src/app/app.module.ts [hunk 1]
```diff
@@ -115,7 +115,8 @@ export const routes: Routes = [
 // Search options (search service)
 export const searchOptions: SearchOptions = {
     routes: ["search"],
-    homeRoute: "home"
+    homeRoute: "home",
+    testDuplicates: (a: Record, b: Record) => a.id === b.id
```
Comment (ericleib) [inline projects/vanilla/src/app/app.module.ts]: tu peux enlever ça, vu qu'on ne va pas gérer les "near duplicates" dans cette app.

**Review by ericleib — APPROVED** [review]

File: projects/vanilla/src/app/app.module.ts (hunk 1 above)
Comment (gleibfried) [reply projects/vanilla/src/app/app.module.ts]: Ca marche !

**Pull request merged**